### PR TITLE
Add Node.js Compat utils implementation

### PR DESCRIPTION
**Pull request by jasnell**
Supported:
* [x] types
* [x] promisify/callbackify
* [x] format/formatWithOptions
* [x] inherits

Questionable (might not implement these, but if we do it will be later)
* inspect
* deprecate
* debug/debuglog
* toUSVString
* MIMEType (experimental added in 19.x)
* parseArgs() (experimental added in 19.x, backported, not relevant?)
* stripVTControlCharacters (not relevant?)
* isDeepStrictEqual

Won't implement:
* transferableAbortController/transferableAbortSignal
* deprecated util.is* methods

### CODE REVIEW

**Comment by jasnell** [comment]
ping @harrishancock @mikea

File: src/node/internal/internal_format.ts [hunk 1]
```diff
@@ -0,0 +1,222 @@
...
+// in all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS
+// OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+// MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN
+// NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM,
+// DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR
+// OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE
+// USE OR OTHER DEALINGS IN THE SOFTWARE.
+
```
Comment (mikea) [inline src/node/internal/internal_format.ts:25]: we'll need eslint disable

File: src/node/util.ts [hunk 2]
```diff
@@ -4,8 +4,182 @@
...
+} from 'node-internal:internal_errors';
+
+export {
+  format
+} from 'node-internal:internal_format';
+
+import { format } from 'node-internal:internal_format';
+
 export const types = internalTypes;
+;
```
Comment (mikea) [inline src/node/util.ts:23]: extra `;`?

File: src/node/util.ts [hunk 3]
```diff
@@ -4,8 +4,182 @@
...
+  }
+  Object.defineProperties(callbackified, descriptors);
+  return callbackified as any;
+}
+
+const kCustomPromisifiedSymbol = Symbol.for('nodejs.util.promisify.custom');
+const kCustomPromisifyArgsSymbol = Symbol('customPromisifyArgs');
+
+// TODO(later): Proper type signature for promisify.
+export function promisify(original: Function): Function {
```
Comment (mikea) [inline src/node/util.ts:66]: I wanted to suggest to copy the one from definitelytyped, but it is huge:

https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/node/util.d.ts#L1050

I'm thinking maybe for node-compat we need to install `@types/node` and use those?

**Review by mikea — APPROVED** [review]

File: src/node/util.ts (hunk 3 above)
Comment (jasnell) [reply src/node/util.ts:66]: Yeah, I had looked there also and quickly ran away.

**Comment by mhart** [comment]
Looks like this was merged with failing tests? I think my PR is failing 'cause of this

**Comment by jasnell** [comment]
Oh interesting hmm.i can have a fix pr shortly but feel free to revert also if necessary.

**Pull request merged**